src/Button/Button.tsx

```diff
@@ -3,49 +3,63 @@ import ButtonVariants from './buttonVariants';
 import IconButton from './IconButton';
 import {
     ButtonComponentProps,
-    COLORS,
     ICON_BUTTON_COLOR,
-    ICON_NAMES,
+    ICON_SVGS,
     InvalidButtonColorError,
     shouldUseMappedIcon
 } from './buttonUtil';
 
 /**
  * Button Component
  *
- * The Button component takes a color and an alt prop, as well as all native button props.
+ * The Button component takes in native button props along with color, alternate, icon, and iconPosition props.
  * The color prop, which defaults to primary, determines its overall color scheme.
- * The alt prop inverts the color scheme to look better on dark backgrounds.
+ * The alternate prop inverts the color scheme to look better on dark backgrounds.
  * className will properly apply itself, but the CSS may need to be more specific,
  * such as button.MyClass instead of just .MyClass.
- *
- * The button can also include an icon configuration to include an icon. The icon configuration includes
- * a name, an optional tooltip, and an optional position. The name, which defaults to 'help', determines which icon to display along side any
- * provided children. If provided, the tooltip will cause the provided text to be displayed
- * when hovering over the button. The position, which defaults to 'left', determines if
- * the icon precedes or proceeds the provided children.
+ * The icon prop can be of several types. If it is an svg, that svg will be used. If it
+ * is strictly "true", we will apply the default icon based on the provided color. In the future, if it is a string,
+ * we will attempt to convert it into an icon. When icon is defined, the position can be specified via the iconPosition prop.
  */
 const Button: FC<ButtonComponentProps> = ({
     alternate = false,
-    color = COLORS.PRIMARY,
+    color = 'primary',
```

Comment:
The type of color is set to `color?: 'primary' | 'secondary' | 'destructive';` which should guide usage to one of the three. I was thinking we could allow for COLORS or the string type, but it would be a bit annoying for consuming applications to HAVE to import our enum to use the Button.

Reply:
Ah good point. My only issue is they pass a string that we do not support. That's kinda what I gleamed when I saw the checking in the button component, if it's a color we support.

Reply:
@michaelpuzon - As Jon mentioned, I noticed when I attempted to make 'color: COLORS' (using the enum), lakehouse threw a big fit in every instance where we had any of the string values used. We can certainly force the user to import the COLORS enum and use one of the values, but that really makes using our component pretty annoying -- at least when typescript is involved.

Reply:
There's some code in there to check if it's a valid color and show a meaningful error about it instead of just crashing:
`if (!ButtonComponent) {
            throw new InvalidButtonColorError(color);
        }`

Reply:
good enough